packages/next/src/server/app-render/use-flight-response.tsx

```diff
@@ -114,15 +113,26 @@ export function createInlinedDataReadableStream(
     async pull(controller) {
       try {
         const { done, value } = await flightReader.read()
-        if (done) {
-          const tail = decoder.decode(value, { stream: false })
-          if (tail.length) {
-            writeFlightDataInstruction(controller, startScriptTag, tail)
+
+        if (value) {
+          try {
+            const decodedString = decoder.decode(value, { stream: !done })
```

Comment:
This part makes me scratch my head. 

I'm not sure its necessary in this instance, but generally, when using `TextDecoder` in stream mode, you should always do one final call to `.decode()` with `stream:false` in order to ensure the final chunk has been decoded. Depending how `flightReader` is implemented, you could have a race condition where `value` is falsey the same time `done` is `false`, and then you may have chunks still inside of the decoder.

Reply:
I'll have to defer to @shuding but we can probably investigate this separate of the reland unless we're aware of specific issue.